### DISCUSSION TITLE
xrdp.service: code=dumped, status =31/SYS

### DISCUSSION THREAD

**Discussion by eliassal** [neutrinolabs/xrdp#3296] [Q&A]
Hi, I have just finished installing a new Fedora 41 VM, when I check the status of xrdp I get 

![image](https://github.com/user-attachments/assets/3f54d8e7-7267-4efb-b7a1-7d10b52193c3)

I am not sur which options are empty
Thanks for your help

**Comment by matt335672** [comment]
@eliassal - I've moved your issue to the right repository and converted it to a discussion as you haven't filled in the issue form.

I've just installed a Fedora 41 workstation VM with xrdp.

I get:-

```
$ sudo systemctl status xrdp
● xrdp.service - xrdp daemon
     Loaded: loaded (/usr/lib/systemd/system/xrdp.service; enabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: active (running) since Mon 2024-11-04 11:39:08 GMT; 3min 50s ago
 Invocation: ddc6d7164aa940e69688f039fb1dcae5
       Docs: man:xrdp(8)
             man:xrdp.ini(5)
   Main PID: 1065 (xrdp)
      Tasks: 1 (limit: 4643)
     Memory: 2.2M (peak: 2.4M)
        CPU: 8ms
     CGroup: /system.slice/xrdp.service
             └─1065 /usr/sbin/xrdp --nodaemon

Nov 04 11:39:08 fedora41 systemd[1]: Starting xrdp.service - xrdp daemon...
Nov 04 11:39:08 fedora41 (xrdp)[1065]: xrdp.service: Referenced but unset environment variable evaluates to an empty string: XRDP_OPTIONS
Nov 04 11:39:08 fedora41 xrdp[1065]: [INFO ] starting xrdp with pid 1065
Nov 04 11:39:08 fedora41 systemd[1]: Started xrdp.service - xrdp daemon.
Nov 04 11:39:08 fedora41 xrdp[1065]: [INFO ] address [0.0.0.0] port [3389] mode 1
Nov 04 11:39:08 fedora41 xrdp[1065]: [INFO ] listening to port 3389 on 0.0.0.0
Nov 04 11:39:08 fedora41 xrdp[1065]: [INFO ] xrdp_listen_pp done
```

The message regarding `XRDP_OPTIONS` is because it is referenced in `/usr/lib/systemd/system/xrdp.service` but not defined. This is fine. You can remove the warning by editing `/etc/sysconfig/xrdp` and uncommenting the `XRDP_OPTIONS` definition.

Your problem is somewhat more concerning. The 31/SYS message indicates that xrdp is trying to access a part of the kernel which it does not have access to. This is a security feature added for #2697 

what do you get for `sudo journalctl | grep 'sig=31'`?

**Comment by eliassal** [comment]
Thanks, I uncommented XRDP_OPTIONS, rebooted the machine but still failing.

Here is the output of the command 

`sudo journalctl | grep 'sig=31'`

```
Nov 01 10:37:18 demosfedora41 audit[3116]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=3116 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7efd637 code=0x0
Nov 01 10:37:18 demosfedora41 audit[3116]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=3116 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 01 12:01:05 demosfedora41 audit[974]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=974 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7f82637 code=0x0
Nov 01 12:01:05 demosfedora41 audit[974]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=974 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 01 13:10:31 demosfedora41 audit[1000]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=1000 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7f74637 code=0x0
Nov 01 13:10:31 demosfedora41 audit[1000]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=1000 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 01 13:12:22 demosfedora41 audit[1253]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=1253 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7efe637 code=0x0
Nov 01 13:12:22 demosfedora41 audit[1253]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=1253 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 01 13:21:41 demosfedora41 audit[2323]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=2323 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7fb8637 code=0x0
Nov 01 13:21:41 demosfedora41 audit[2323]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=2323 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 02 10:35:07 demosfedora41 audit[967]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=967 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7fdf637 code=0x0
Nov 02 10:35:07 demosfedora41 audit[967]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=967 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 02 11:16:02 demosfedora41 audit[996]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=996 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7f03637 code=0x0
Nov 02 11:16:02 demosfedora41 audit[996]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=996 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 02 12:33:33 demosfedora41 audit[966]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=966 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7efe637 code=0x0
Nov 02 12:33:33 demosfedora41 audit[966]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=966 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 02 13:01:33 demosfedora41 audit[945]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=945 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7f14637 code=0x0
Nov 02 13:01:33 demosfedora41 audit[945]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=945 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 02 17:57:00 demosfedora41 audit[969]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=969 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7f33637 code=0x0
Nov 02 17:57:00 demosfedora41 audit[969]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=969 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 02 18:11:00 demosfedora41 audit[966]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=966 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7fa8637 code=0x0
Nov 02 18:11:00 demosfedora41 audit[966]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=966 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 02 18:17:40 demosfedora41 audit[942]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=942 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7fa6637 code=0x0
Nov 02 18:17:40 demosfedora41 audit[942]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=942 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
Nov 04 13:17:02 demosfedora41 audit[922]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=922 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 arch=40000003 syscall=45 compat=1 ip=0xf7f5d637 code=0x0
Nov 04 13:17:02 demosfedora41 audit[922]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=system_u:system_r:unconfined_service_t:s0 pid=922 comm="xrdp" exe="/usr/sbin/xrdp" sig=31 res=1
```

**Comment by matt335672** [comment]
Thanks

`syscall=45` translates to `recvfrom`:-

```
$ scmp_sys_resolver 45
recvfrom
```

According to `systemd-analyze syscall-filter`, `recvfrom` is in the set `@network-io`, and that is in `@system-service`. That's how xrdp should be configured.

What do you get for `systemctl cat xrdp.service`?

**Comment by eliassal** [comment]
Not sure I understand what I should do with the command scmp_sys_resolver 45 !

 Here is the content for xrdp service

```
# /usr/lib/systemd/system/xrdp.service
[Unit]
Description=xrdp daemon
Documentation=man:xrdp(8) man:xrdp.ini(5)
Requires=xrdp-sesman.service
After=network-online.target xrdp-sesman.service

[Service]
Type=exec
EnvironmentFile=-/etc/sysconfig/xrdp
EnvironmentFile=-/etc/default/xrdp
ExecStart=/usr/sbin/xrdp $XRDP_OPTIONS --nodaemon
SystemCallArchitectures=native
SystemCallFilter=@system-service

[Install]
WantedBy=multi-user.target

# /usr/lib/systemd/system/service.d/10-timeout-abort.conf
# This file is part of the systemd package.
# See https://fedoraproject.org/wiki/Changes/Shorter_Shutdown_Timer.
#
# To facilitate debugging when a service fails to stop cleanly,
# TimeoutStopFailureMode=abort is set to "crash" services that fail to stop in
# the time allotted. This will cause the service to be terminated with SIGABRT
# and a coredump to be generated.
#
# To undo this configuration change, create a mask file:
#   sudo mkdir -p /etc/systemd/system/service.d
#   sudo ln -sv /dev/null /etc/systemd/system/service.d/10-timeout-abort.conf

[Service]
TimeoutStopFailureMode=abort

```

**Comment by matt335672** [comment]
The `scmp_sys_resolver` command just translates the `syscall=` number into the name of the service.

This is a mystery. According to `xrdp.service`, `xrdp` is allowed to use the `recvfrom` call, but the kernel disagrees.

Have you rebooted, or tried `sudo systemctl daemon-reload`?

**Comment by eliassal** [comment]
I have rebooted so many times

**Comment by matt335672** [comment]
OK - thanks for clarifying that.

There's a mismatch between my system and yours, clearly. Let's try to attack this from both ends.

1) What do you get on your system for `scmp_sys_resolver 45`? You may need to install the `libseccomp-devel` package if it isn't already installed. Maybe on your system this maps to another system service.
2) Have you made any changes as all to `xrdp.ini`? If so, please can you post the entire file, so I can try to reproduce your problem here.

**Comment by eliassal** [comment]
HI, here is the output for scmp_sys_resolver 45
![image](https://github.com/user-attachments/assets/8e03d170-c45b-4c67-8aa5-3818450dab76)

I have touched nothing, this is a fresh new fedora 41 with xfce image install (downloaded from fedora downloads), enclosed the file (I changed ini to txt) so I can upload it
[xrdp.txt](https://github.com/user-attachments/files/17634153/xrdp.txt)

**Comment by matt335672** [comment]
That all looks fine.

The only difference in your file is that the `[Xorg]` stanza is uncommented in yours. I've uncommented mine, installed `xorgxrdp`, and I'm still having no problems.

I've just done a `dnf update`, and all is still well. Can you do the same, and then post your package list with `rpm -qa | sort` ?

**Comment by matt335672** [comment]
Also, at what stage does xrdp fail? Does it fail immediately on running?

**Comment by eliassal** [comment]
It fails right away. I did install xorgxrdp restarted the VM getting same behavior

![image](https://github.com/user-attachments/assets/69aa2965-7ece-4735-a314-f720db78966e)

Tell me please, did you install the same Fedora 41 with xfce?
Here is the package list file.
[packageList.txt](https://github.com/user-attachments/files/17645641/packageList.txt)

**Comment by matt335672** [comment]
No, I didn't install the xfce version.

I think I've found the problem however.

In your package list:-

<pre>
xrdp-0.10.1-1.fc41.<b>i686</b>
</pre>

You've got the 32-bit version of xrdp installed on a 64-bit system.

I've just installed the 32-bit version of xrdp on my system and I can reproduce the fault.

Try removing it and installing the 64-bit package (i.e.):-

```
sudo dnf remove xrdp.i686
sudo dnf install xrdp.x86_64
sudo systemctl enable --now xrdp
systemctl status xrdp
```

**Comment by eliassal** [comment]
Correct, fantatstic, you are really the man to ask. Now xrdp is active and not failing but when I try to RDP I get a black screen and then disconnects right away.
I uncommented xorg in xrdp.ini, when I try to RDP I get first

![image](https://github.com/user-attachments/assets/f4f65990-4278-4707-95f2-16485d7edf5f)

Then in a 2nd screen, I eneter my password

![image](https://github.com/user-attachments/assets/cc908708-c1e6-4f05-98ca-ee784263827f)

and get disconnected right away

**Comment by eliassal** [comment]
Hurray, I had a session connected directly within hyper-v since I was not able to RDP. I closed it and all of a sudden I was able to RDP to the machine.
Sop does this mean that I cant have 2 sessions, one direct one RDP?
So many thanks for your help

**Comment by matt335672** [comment]
With systemd-based distros you can't have more than one graphical session per user logged in at once. This is covered in [the FAQ](https://github.com/neutrinolabs/xrdp/wiki/Tips-and-FAQ#why-cant-i-log-the-same-user-on-on-the-graphical-console-and-over-xrdp-at-the-same-time)

The simplest workaround is to use a different user for RDP-based sessions.

**Comment by eliassal** [comment]
Thanks matt for the explanation. Last thing, yesterday when I RDPed I was able to copy/paste from/to VM to my host but this stopped this morninng nothing happens in the 2 ways. What might happened? I use mobaxterm to RDP

**Comment by matt335672** [comment]
A couple of possibilities:-
1) There's a problem at the moment where if you've already got a client connected from one location, and you connect another one, the session handover doesn't work correctly. That might be it.
2) cut/paste is handled by the `xrdp-chansrv` process running in the user session. Check that's still running.
